lib/ButtonNew/ButtonIcon/ButtonIconDark.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import cx from 'classnames';
+import { Icon } from 'lib';
+import { ButtonBase } from '../ButtonBase';
+import {
+  sizes,
+  getSizeClasses,
+  buttonIconPropTypes,
+  buttonIconDefaultProps
+} from '../common';
+
+function ButtonIconDark({
+  className,
+  disabled,
+  name,
+  active,
+  size,
+  iconTitle,
+  children,
+  ...rest
+}) {
+  const nonDisabledClasses = cx(
```

Comment:
🔴  I've mentioned this in #884 but I'm not digging how we're implementing tailwind in these scenarios.

After come back and trying to work with `ButtonIcon` it's almost impossible to understand what is what. This ends up with whoever makes future changes, either bolting on another prop or causing bugs because they forget something like the focus state.

Here are my main issues;

- The implementation isn't easy to digest for those who didn't write it.
- It's CSS which is the most readable programming syntax (for the most part) and this isn't so why is that?

Especially with how `classnames` works, you specify a string for classes and then the predicate after so I have to parse an entire long string before I can understand when it is applied.

And that's before prettier comes along and starts breaking things up making it even worse.

----

I believe you've suggested we pair on #884 which I'm down for.

Reply:
Let's pair! Perhaps we can bring this up in front end catch up today too, while I don't have a problem with this implementation, there is a card for making our button styles into tailwind components that might solve your problem.